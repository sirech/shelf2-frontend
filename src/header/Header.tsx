// @flow

import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { LinkContainer } from 'react-router-bootstrap'
import { Helmet } from 'react-helmet'

import Logout from './Logout'
import SearchBar from './SearchBar'

import {
  Collapse,
  Nav,
  Navbar,
  NavbarBrand,
  NavItem,
  NavbarToggler,
} from 'reactstrap'

const Login = () => (
  <NavItem>
    <Link to="/login" className="nav-link">
      Login
    </Link>
  </NavItem>
)

const Header = () => {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <header className="mb-4">
      <Helmet>
        <meta charSet="utf-8" />
        <title>Shelf</title>
      </Helmet>
      <Navbar dark color="dark" expand="sm">
        <LinkContainer to="/books">
          <NavbarBrand>Shelf</NavbarBrand>
        </LinkContainer>
        <NavbarToggler onClick={() => setIsOpen(!isOpen)} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <SearchBar />
            <Login />
            <Logout />
          </Nav>
        </Collapse>
      </Navbar>
    </header>
  )
}

export default Header
