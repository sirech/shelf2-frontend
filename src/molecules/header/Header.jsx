import React from 'react'
import { Link } from 'react-router-dom'
import { LinkContainer } from 'react-router-bootstrap'
import { Helmet } from 'react-helmet'

import { Collapse, Nav, Navbar, NavbarBrand, NavItem, NavbarToggler } from 'reactstrap'

class Header extends React.Component {
  constructor (props) {
    super(props)

    this.toggle = this.toggle.bind(this)
    this.state = {
      isOpen: false
    }
  }

  toggle () {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  render () {
    return (
      <header className='mb-4'>
        <Helmet>
          <meta charSet='utf-8' />
          <title>Shelf</title>
        </Helmet>
        <Navbar inverse color='inverse' toggleable>
          <NavbarToggler right onClick={this.toggle} />
          <LinkContainer to='/books' >
            <NavbarBrand>Shelf</NavbarBrand>
          </LinkContainer>
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className='ml-auto' navbar>
              <NavItem>
                <Link to='/login' className='nav-link'>Login</Link>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </header>
    )
  }
}

export default Header
