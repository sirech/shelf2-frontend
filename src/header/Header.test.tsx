import React from 'react'
import { screen } from '@testing-library/react'

import Header from './Header'
import { fullRender } from 'test'

describe('components', () => {
  describe('Header', () => {
    it('renders correctly', async () => {
      fullRender(<Header />)

      await screen.findByText('Shelf')
    })
  })
})
