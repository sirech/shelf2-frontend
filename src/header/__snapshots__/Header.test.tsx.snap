// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header renders correctly 1`] = `
<header
  className="mb-4"
>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/books"
      onClick={[Function]}
    >
      Shelf
    </a>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <form
          className="form-inline"
        >
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
          />
        </form>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <div
            className="nav-link"
            onClick={[Function]}
            role="link"
          >
            Logout
          </div>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
